src/modeler/AutoRest.Swagger/Validation/BodyTopLevelProperties.cs

```diff
@@ -28,62 +28,25 @@ public class BodyTopLevelProperties : TypedRule<Dictionary<string, Operation>>
+            foreach (var violatingModel in violatingModels)
             {
-                if ((operation.ToLower().Equals("get") ||
-                    operation.ToLower().Equals("put") ||
-                    operation.ToLower().Equals("patch")) && path[operation]?.Parameters != null)
-                {
-                    foreach (SwaggerParameter param in path[operation].Parameters)
-                    {
-                        if (param.In == ParameterLocation.Body)
-                        {
-                            if (param?.Schema?.Reference != null)
-                            {
-                                string defName = Extensions.StripDefinitionPath(param.Schema.Reference);
-                                var definition = ((ServiceDefinition)context.Root).Definitions[defName];
-                                if (definition?.AllOf != null && definition.Properties != null &&
-                                    definition.AllOf.Select(s => s.Reference).Where(reference => resourceRefRegEx.IsMatch(reference)) != null)
-                                {
-                                    //Model is allOf Resource
-                                    foreach (KeyValuePair<string, Schema> prop in definition.Properties)
-                                    {
-                                        if (!allowedTopLevelProperties.Contains(prop.Key.ToLower()))
-                                        {
-                                            notAllowedProperties.Add(defName + "/" + prop.Key);
-                                        }
-                                    }
-                                }    
-                            }
-                            if (param?.Schema?.AllOf != null && param.Schema.Properties != null &&
-                                param.Schema.AllOf.Select(s => s.Reference).Where(reference => resourceRefRegEx.IsMatch(reference)) != null)
-                            {
-                                //Model is allOf Resource
-                                foreach (KeyValuePair<string, Schema> prop in param.Schema.Properties)
-                                {
-                                    if (!allowedTopLevelProperties.Contains(prop.Key.ToLower()))
-                                    {
-                                        notAllowedProperties.Add(path[operation].OperationId + ":" + prop.Key);
-                                    }
-                                }
-                            }
-                          }  
-                        }
-                    }
-                }
-            formatParameters = new[] { string.Join(", ", notAllowedProperties.ToArray()) };
-            return (notAllowedProperties.Count() == 0);
+                yield return new ValidationMessage(new FileObjectPath(context.File, context.Path.AppendProperty(violatingModel).AppendProperty("properties")), this, 
```

Comment:
is the path to the object missing the actual property violating the rule? could you paste an example output?